lib/internal/errors.js

```diff
@@ -64,6 +66,19 @@ const prepareStackTrace = (globalThis, error, trace) => {
     return f(error, trace);
   }
 
+  const firstFrame = trace[0]?.getFunctionName();
+  if (firstFrame && firstFrame.startsWith(nodeInternalPrefix)) {
+    for (let l = trace.length - 1; l >= 0; l--) {
+      const fn = trace[l].getFunctionName();
+      if (fn != null && fn.startsWith(nodeInternalPrefix)) {
+        trace.splice(0, l + 1);
+        break;
+      }
+    }
```

Comment:
That value is always updated before calling `new Error()` from Node. And when processing an exception originating inside user code, that part of the function does not run.
`Error.stackTraceLimit` is modified to be `Infinity` when processing Node exceptions - so that the full stack can be examined - then removing Node internals and splicing it to the correct size, before restoring the user value at the very end of `captureLargerStackTrace`

Reply:
OK, could you add a comment explaining that?

Reply:
That part of the code has been changed, what is this referring to?

Reply:
I think it could be useful to add a comment explaining why you are using a cached value for `userStackTraceLimit` (what you explained in https://github.com/nodejs/node/pull/35644/files#r519655249 basically).